lib/rubygems/platform.rb

```diff
@@ -146,10 +146,17 @@ def hash # :nodoc:
   ##
   # Does +other+ match this platform?  Two platforms match if they have the
   # same CPU, or either has a CPU of 'universal', they have the same OS, and
-  # they have the same version, or either has no version.
+  # they have the same version, or either one has no version
   #
   # Additionally, the platform will match if the local CPU is 'arm' and the
   # other CPU starts with "arm" (for generic ARM family support).
+  #
+  # Of note, this method is not commutative. Indeed the OS 'linux' has a
+  # special case: the version is the libc name, yet while "no version" stands
+  # as a wildcard for a binary gem platform (as for other OSes), for the
+  # runtime platform "no version" stands for 'gnu'. To be able to disinguish
+  # these, the method receiver is the gem platform, while the argument is
+  # the runtime platform.
```

Comment:
Given the goal we had, and the fact that `===` is not commutative already in many places (e.g class membership) this is how I implemented the constraint.

Therefore calls to `Gem::Platform#===` should always be in that order: `gem_platform === runtime_platform`, turns out this was already the case almost everywhere.

Reply:
[While doing this](https://github.com/rubygems/rubygems/issues/3174#issuecomment-963166985) it turns out I found other places where this test is called in a certain order, so I [reversed the order](https://gist.github.com/lloeki/12dcf61324f64a2fa1e8a8b2109c1f00#file-musl-patch-L54).

Reply:
As already observed, `Gem::Platform#===` is already not commutative so your implementation seems fine for the time being!